packages/workbox-google-analytics/index.mjs

```diff
@@ -0,0 +1,172 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+
+/**
+ * @module workbox-google-analytics
+ */
+
+import {Queue, QueuePlugin} from 'workbox-background-sync';
+import {Route, Router} from 'workbox-routing';
+import {NetworkFirst, NetworkOnly} from 'workbox-runtime-caching';
+import {
+  CACHE_NAME,
+  QUEUE_NAME,
+  MAX_RETENTION_TIME,
+  GOOGLE_ANALYTICS_HOST,
+  ANALYTICS_JS_PATH,
+  COLLECT_PATH,
+} from './lib/constants.mjs';
+import './_version.mjs';
+
+/**
+ * Promisifies the FileReader API to await a text response from a Blob.
+ *
+ * @private
+ * @param {Blob} blob
+ * @return {Promise<string>}
+ */
+const getTextFromBlob = (blob) => {
```

Comment:
What do you think about putting this into a separate `.mjs` file? That would make it easier to write a unit test for it.

Reply:
It's hard to test as is since it'd basically just be testing our stubs, but it's tested pretty well implicitly (as an implementation detail) via the rest of the tests in this package.

I think it's fine for now, but if another package needs this functionality then it makes sense to move to workbox-core and add explicit tests for it.